### PR TITLE
add CI scripts

### PR DESCRIPTION
**Pull request by takam5f2**
Markdownに対してCIを追加します。

### CODE REVIEW

File: .github/workflows/generate_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: generate-docs
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "mkdocs.yaml"
```
Comment (kenji-miyake) [inline .github/workflows/generate_docs.yaml:8]: Please rename here or the file name.
https://github.com/tier4/CARET_doc/blob/main/mkdocs.yml

File: .prettierrc.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+printWidth: 120
```
Comment (kenji-miyake) [inline .prettierrc.yaml:1]: FYI we changed here to 100 because ROS2's coding guideline is so.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+ci:
+  autofix_commit_msg: "ci(pre-commit): autofix"
+  autoupdate_commit_msg: "ci(pre-commit): autoupdate"
```
Comment (kenji-miyake) [inline .pre-commit-config.yaml:3]: You can activate pre-commit CI here, which can automatically fix errors.
https://pre-commit.ci/

File: .markdownlint.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+default: true
+MD013: false
+MD024:
+  siblings_only: true
+MD033: false
+MD041: false
```
Comment (kenji-miyake) [inline .markdownlint.yaml:6]: Adding MD046 is good if you use `admonition`.
https://github.com/markdownlint/markdownlint/issues/215

File: .github/workflows/generate_docs.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      - "**/*.md"
+      - "**/*.svg"
+      - "**/*.png"
+      - "**/*.css"
+
+jobs:
+  get-docs-version-name:
+    uses: autowarefoundation/autoware-github-actions/.github/workflows/reusable-get-docs-version-name.yaml@tier4/proposal
```
Comment (kenji-miyake) [inline .github/workflows/generate_docs.yaml:23]: We'll probably replace these workflows with `composite`. So please keep in mind that.

File: .github/workflows/generate_docs.yaml (hunk 1 above)
Comment (takam5f2) [reply .github/workflows/generate_docs.yaml:8]: Thank you for pointing out so, but I'd like to rename `mkdocs.yml` as `mkdocs.yaml`.

File: .prettierrc.yaml (hunk 2 above)
Comment (takam5f2) [reply .prettierrc.yaml:1]: I'd like to fix it as you told.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (takam5f2) [reply .pre-commit-config.yaml:3]: Thank you! I'd like to add the following.
```
    autoupdate_schedule: weekly
```

File: .markdownlint.yaml (hunk 4 above)
Comment (takam5f2) [reply .markdownlint.yaml:6]: I'm adding it.

File: .github/workflows/generate_docs.yaml (hunk 5 above)
Comment (takam5f2) [reply .github/workflows/generate_docs.yaml:23]: I'd like to make issue for my reminder. Thanks!

File: .pre-commit-config.yaml (hunk 3 above)
Comment (kenji-miyake) [reply .pre-commit-config.yaml:3]: It's okay, but I think it's not necessary since it's a default value.

File: .markdownlint.yaml (hunk 4 above)
Comment (kenji-miyake) [reply .markdownlint.yaml:6]: f8405941a353c96e23eeef2a2e3545b705ad3891

File: .prettierrc.yaml (hunk 2 above)
Comment (kenji-miyake) [reply .prettierrc.yaml]: f8405941a353c96e23eeef2a2e3545b705ad3891

File: .github/workflows/generate_docs.yaml (hunk 1 above)
Comment (kenji-miyake) [reply .github/workflows/generate_docs.yaml:8]: f8405941a353c96e23eeef2a2e3545b705ad3891

File: .pre-commit-config.yaml (hunk 3 above)
Comment (takam5f2) [reply .pre-commit-config.yaml:3]: Thank you, and removed default value.  d3b8812d41e27120669e77d0a298e543378c9d11

**Comment by takam5f2** [comment]
@hsgwa   
お時間あるときにこちら確認お願いします。
スペルチェックエラーが出ていますが、別のPRで対応するで良いかと思います。

**Comment by hsgwa** [comment]
ありがとうございます。確認したのでマージ致します。

**Review by hsgwa — APPROVED** [review]

**Pull request merged**